backend/src/main/java/ch/puzzle/okr/service/business/AlignmentBusinessService.java

```diff
@@ -0,0 +1,131 @@
+package ch.puzzle.okr.service.business;
+
+import ch.puzzle.okr.ErrorKey;
+import ch.puzzle.okr.exception.OkrResponseStatusException;
+import ch.puzzle.okr.models.Objective;
+import ch.puzzle.okr.models.alignment.Alignment;
+import ch.puzzle.okr.models.alignment.KeyResultAlignment;
+import ch.puzzle.okr.models.alignment.ObjectiveAlignment;
+import ch.puzzle.okr.models.keyresult.KeyResult;
+import ch.puzzle.okr.service.persistence.AlignmentPersistenceService;
+import ch.puzzle.okr.service.persistence.KeyResultPersistenceService;
+import ch.puzzle.okr.service.persistence.ObjectivePersistenceService;
+import ch.puzzle.okr.service.validation.AlignmentValidationService;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class AlignmentBusinessService {
+
+    private final AlignmentPersistenceService alignmentPersistenceService;
+    private final AlignmentValidationService alignmentValidationService;
+    private final ObjectivePersistenceService objectivePersistenceService;
+    private final KeyResultPersistenceService keyResultPersistenceService;
+
+    public AlignmentBusinessService(AlignmentPersistenceService alignmentPersistenceService,
+            AlignmentValidationService alignmentValidationService,
+            ObjectivePersistenceService objectivePersistenceService,
+            KeyResultPersistenceService keyResultPersistenceService) {
+        this.alignmentPersistenceService = alignmentPersistenceService;
+        this.alignmentValidationService = alignmentValidationService;
+        this.objectivePersistenceService = objectivePersistenceService;
+        this.keyResultPersistenceService = keyResultPersistenceService;
+    }
+
+    public String getTargetIdByAlignedObjectiveId(Long alignedObjectiveId) {
+        alignmentValidationService.validateOnGet(alignedObjectiveId);
+        Alignment alignment = alignmentPersistenceService.findByAlignedObjectiveId(alignedObjectiveId);
+        if (alignment instanceof KeyResultAlignment keyResultAlignment) {
+            return "K" + keyResultAlignment.getAlignmentTarget().getId();
+        } else if (alignment instanceof ObjectiveAlignment objectiveAlignment) {
+            return "O" + objectiveAlignment.getAlignmentTarget().getId();
+        } else {
+            return null;
```

Comment:
Is ok to be null, because not every objective has an alignment, then in frontend, it does not need to load any alignment into dropdown in objective form